Sprint 5/step 1

AJIEKCEY opened the pull request.

AJIEKCEY committed:
sprint-5/step-1
Добавить ленту заказов.
Добавить поддержку вебсокетов

AJIEKCEY committed:
sprint-5/step-1
Добавить карточку заказа.

AJIEKCEY committed:
sprint-5/step-1
Добавить компоненты для ленты заказов профиля.
Добавить получение заказов.
Поправить стили.

AJIEKCEY committed:
sprint-5/step-1
Исправить проверку пользователя.
Добавить загрузку ингредиентов для профиля.
Добавить сортировку в заказы пользователя.

gennady-bars commented on src/components/profile-orders/profile-orders.tsx:
```diff
@@ -0,0 +1,80 @@
+	useEffect(() => {
+		// Проверяем, загружены ли ингредиенты
+		if (ingredients.length === 0) {
+			dispatch(fetchIngredients());
```
`dispatch(fetchIngredients())` достаточно 1 раз вызвать в `App`, чтобы все страницы получили ингредиенты при загрузке сайта. Не нужно дублировать запрос в других компонентах.

AJIEKCEY replied:
Изменил

gennady-bars commented on src/components/feed-stats/feed-stats.tsx:
```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+import { useSelector } from 'react-redux';
```
* Хуки `useSelector` и `useDispatch` нужно импортировать из файла `redux.ts` (они там типизированные уже), а не из библиотеки `react-redux`.  Нужно найти через поиск `react-redux` и заменить **все импорты**.

AJIEKCEY replied:
Поправил. Спасибо!

gennady-bars commented on src/components/order-card/order-card.tsx:
```diff
@@ -0,0 +1,136 @@
+
+	const handleClick = () => {
+		// Используем номер заказа в URL, а не ID
+		navigate(`/feed/${order.number}`, {
```
* При открытии попапа в **истории заказов** отображается путь `feed`, хотя должен `/profile/orders` быть https://disk.yandex.ru/i/izl0Yqc-3vkztQ

AJIEKCEY replied:
Исправил.

gennady-bars reviewed (commented):
Здравствуйте. (**Нужно развернуть общий комментарий ↓**)

Посмотрите на **гитхабе** все комментарии к коду (нужно прокрутить вниз страницу там)

Работа проделана огромная:



* Отлично типизировали хуки `useSelector` и `useDispatch`
* Отлично типизировали стейт, диспатч 

но есть некоторые недочеты:



* Хуки `useSelector` и `useDispatch` нужно импортировать из файла `redux.ts` (они там типизированные уже), а не из библиотеки `react-redux`.  Нужно найти через поиск `react-redux` и заменить **все импорты**.
* При открытии попапа в **истории заказов** отображается путь `feed`, хотя должен `/profile/orders` быть https://disk.yandex.ru/i/izl0Yqc-3vkztQ

AJIEKCEY committed:
sprint-5/step-1
Вынести загрузку ингредиентов в app.tsx

AJIEKCEY committed:
sprint-5/step-1
Поправить импорты для useAppSelector и useAppDispatch

AJIEKCEY committed:
sprint-5/step-1
Исправить url для карточки заказа в истории заказов.
Поправить работу с webSocket

AJIEKCEY committed:
sprint-5/step-1
Исправления после code analysis

gennady-bars reviewed (commented):
Еще немного, и все будет отлично:

* Не открывается попап с номером заказа после успешного оформления заказа https://disk.yandex.ru/i/esNkjDe_xN9kSQ 
* При прямом переходе по ссылке на заказ **истории заказов** не отображается **страница** с деталями заказа

AJIEKCEY committed:
sprint-5/step-1
Исправить отображение модального окна после заказа.
Исправить отображение страницы заказа в истории заказа.

gennady-bars reviewed (approved):
Поздравляю! Ваша работа принята.

Вы отлично потрудились.

Удачного дальнейшего обучения.
